### PR TITLE
[FIX] website: remove 'is_homepage' from menu node

### PR DESCRIPTION
**Pull request by kea14**
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website/static/src/js/menu/content.js [hunk 1]
```diff
@@ -568,6 +568,7 @@ var EditMenuDialog = weWidgets.Dialog.extend({
         node.children.forEach(function (child) {
             self._flatenize(child, _dict);
         });
+        delete node['is_homepage'];
```
Comment (qsm-odoo) [inline addons/website/static/src/js/menu/content.js]: @kea14 How can you come with this fix after an afternoon of investigation **and** the help of romain...

- The `_flatenize` method, as described, returns a mapping id -> menu item containing all the menu items in the given menu hierarchy... how can you find ok to modify in place the given `node` argument. When I read this function with your change, at best it does nothing because `node` is a temp object in the caller and at worst you break everything since you modify an object the caller does not want to be modified.

- Why would want to delete the received `is_homepage` info at all?? Don't you think we fetch it from the server for a reason? Don't you ask yourself how it is used before removing it? After 5sec investigation, I see that the `is_homepage` seems always `null` since 12.0 and that it is something to be fixed to show the "home" icon in the dialog.

- Even so... even with your branch, I still get tracebacks with simple editions of a menu hierarchy. How did you test that you restored the feature...

Comment (kea14) [reply addons/website/static/src/js/menu/content.js]: Ok, it was probably clumsy to do that. When talking with RDE this is what I understood (and even if indeed what I did was wrong, this is what we precisely discussed and got to `_flatenize` fn) but ok and in any way I have to admit my mistake on that.

> After 5sec investigation, I see that the is_homepage seems always null since 12.0

Yes this is what I noticed yesterday and this is where I did not put enough attention on as this is needed for the homepage icon. But I want to come back on one thing here. Indeed it is the case since 12.0 (or even before) but it caused a warning until today as it is causing now an error probably due to an ORM change or whatever. The ORM tries to write on `is_homepage` field in `website.menu` but it does not exist.

> I still get tracebacks with simple editions of a menu hierarchy.

What are you talking about ? Ok, Because of what I did there is no more homepage icon (which is dumb) but what are the tracebacks are you talking about ? In THIS branch when saving Edit Menu dialog, no traceback. 

Concerning the branch `saas-12.5-web-editor-new-ui-add-mega-menu-aul` you still get a traceback when you hit save in this Edit Menu dialog as the fix is potentially in THIS branch here and not in the other. Just tell me if it needs to be done differently.

So basically, I could test the "restored feature" when I added a mega menu. What is so unclear ? And how could I talk about the issue that I have with `dropdown.js` stuff if I hadn't tested ?
This issue in Edit Menu dialog is not related to my branch but on the saas-12.5 branch. 

Common, I can understand your displeasure but there is no need to be unpleasant.
So, if you could check if this is done in a correct way now, that would be nice :smiley: Thanks a lot!

Comment (qsm-odoo) [reply addons/website/static/src/js/menu/content.js]: > In THIS branch when saving Edit Menu dialog, no traceback.

Make a sub-menu, edit, save, traceback... it does not seem to be a complex use case I did when testing THIS branch :disappointed: So, like I tell you each review: when I tell you I tested and had a traceback with a basic flow, I had a traceback with a basic flow. Clicking Edit -> Save is not testing enough to say you restored the feature of editing the whole menu.

> Concerning the branch `saas-12.5-web-editor-new-ui-add-mega-menu-aul` you still get a traceback when you hit save in this Edit Menu dialog as the fix is potentially in THIS branch here and not in the other.

Well yes, your mega menu branch should be rebased on top of your fixes. But I was not talking about the mega menu branch anyway.

Comment (kea14) [reply addons/website/static/src/js/menu/content.js]: Well... I believe you but I've just tried again (and not only clicking Edit -> Save like you said) and I have nothing (local or runbot). Making a sub-menu, edit, save, making a menu, edit, save, deleting menu, save, making new menu and new sub-menu, save, etc......
Could you give it another try ? So I can view the problem and fix this

**Comment by qsm-odoo** [comment]
Continuing on https://github.com/odoo/odoo/pull/36204
